src/main/java/io/openio/sds/AdvancedClient.java

```diff
@@ -193,6 +254,23 @@ public void setContainerProperties(OioUrl url, Map<String, String> props,
     public Map<String, String> getContainerProperties(OioUrl url, RequestContext reqCtx)
             throws OioException;
 
+    /**
+     * Retrieve user properties and system properties
+     * of the specified container.
+     *
+     * @param url
+     *            the url of the container
+     * @param reqCtx
+     *            common parameters to all requests
+     * @return the user properties and system properties found on the container
+     * @throws ContainerNotFoundException
+     *             if the specified container doesn't exist
+     * @throws OioSystemException
+     *             if any error occurs during request execution
+     */
+    public Map<String, Map<String, String>> getAllContainerProperties(OioUrl url,
+            RequestContext reqCtx) throws OioException;
```

Comment:
Returning a map which will ever contain only 2 values seems overkill. I suggest returning an array of `Map<string, String>` (or maybe a list?).

Reply:
With `JsonUtils.jsonToMapMap` (https://github.com/open-io/oio-api-java/pull/71/files#diff-f0a51d0012e81f76617778014e7d65aeR1028), we have automatically `Map<string, Map<string, String>>`. I do not think it's useful to convert a `Map<string, Map<string, String>>` to a `List<Map<string, String>>`.

Reply:
Ok.